events/syscall/read_ssh_information.go

```diff
@@ -0,0 +1,48 @@
+//go:build linux
+// +build linux
+
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2023 The Falco Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package syscall
+
+import (
+	"github.com/falcosecurity/event-generator/events"
+
+	"os"
+)
+
+var _ = events.Register(
+	ReadSshInformation,
+	events.WithDisabled(), // the rule is not enabled by default, so disable the action too
+)
+
+func ReadSshInformation(h events.Helper) error {
+	// Creates temporary data for testing.
+	directoryname := "/home/created-by-falco-event-generator/.ssh"
+	if err := os.MkdirAll(directoryname, 0755); err != nil {
+		return err
+	}
+
```

Comment:
```suggestion
defer os.RemoveAll("/home/created-by-falco-event-generator") 
```

Reply:
In this way, even if below `WriteFile` fails, we still cleanup the home folder.
Also, i don't really like the fact that we might truncate a `created-by-falco-event-generator` user's `known_hosts` file with this event...

Reply:
We might generate an uuid and check beforehand if `/home/foo` exists before starting the event.
Perhaps a small utility function (that will be probably used at least by #117).
Ie: something like this (pseudo code):
```
var (
  homeDir string
  err error
)
for {
  if homeDir, err = os.MkdirTemp(homeDir, "falco-event-generator-"); err == nil {
    break
  }
}
sshDir := homeDir + ".ssh"
...
}
```

Reply:
https://github.com/falcosecurity/event-generator/blob/27dda87a555215942cc8ec093691dae4fb15ce93/events/syscall/read_ssh_information.go#L34-L37

Hey @FedeDP here `os.MkdirAll(directoryname, 0755)` will create directory if it wont exist and if directory already exists it returns nil. So before starting event we are checking "/home/created-by-falco-event-generator" exists or not and if not we create one.

This is what we want to achieve right?

Reply:
Ok got it! You are saying to make an utility function which takes directory name as input!

and in function it checks whether directory exists or not , If not then create one. Am I  right?

Reply:
> Ok got it! You are saying to make an utility function which takes directory name as input!

Basically what will happen if an user named `created-by-falco-event-generator` exists in my system? `os.MkdirAll` returns nil and we will then gladly proceed to wipe `known_hosts` file; see https://pkg.go.dev/os#MkdirAll: 
> If path is already a directory, MkdirAll does nothing and returns nil. 

What i am proposing is to generate a temp folder under `"/home"` until it succeeds. `os.MkdirTemp` differs from `MkdirAll` since it returns an error in case the folder we attempted to create is already existent.

Reply:
Please note that the `for` loop is not necessary, since `os.MkdirTemp` does that internally:
https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/os/tempfile.go;l=95

Reply:
Yep Got it! Just did the changes.